File: wau-api/src/main/java/com/xegami/wau/api/controller/RoomMessage.java
package com.xegami.wau.api.controller;

import com.xegami.wau.api.dto.PlayerDTO;
import com.xegami.wau.api.dto.RoomDTO;
import lombok.Data;

@Data
public class RoomMessage {

    private PlayerDTO agent;
    private Integer code;
    private String password;
    private PlayerDTO target;
    private boolean afk;
    private boolean rejoining;
    private boolean kicked;
    private boolean hotMode;
    private String roomToken;
    private Integer roundNumber;

}

File: wau-android/app/src/main/java/com/xegami/wau/android/rest/controller/RoomMessage.kt
package com.xegami.wau.android.rest.controller

import com.google.gson.Gson
import com.xegami.wau.android.rest.dto.PlayerDTO

data class RoomMessage(val agent: PlayerDTO, val code: Int = -1, val password: String = "", val target: PlayerDTO? = null, val afk: Boolean = false, val rejoining: Boolean = false, val kicked: Boolean = false, val hotMode: Boolean = false, val roomToken: String? = null, val roundNumber: Int = -1) {

    fun toJson(): String {
        return Gson().toJson(this)
    }
}